### PR TITLE
Remove deprecations introduced in 2.4.x

### PR DESCRIPTION
**Pull request by jonas**
Refs #1645

### CODE REVIEW

**Comment by raboof** [comment]
(looks like the jenkins failure is MiMa)

File: akka-http-core/src/main/java/akka/http/javadsl/model/HttpMethod.java [hunk 1]
```diff
@@ -42,14 +42,13 @@ public final String name() {
 
     /**
      * Returns the entity acceptance level for this method.
-     * @deprecated Use {@link #getRequestEntityAcceptance} instead, which returns {@link akka.http.javadsl.model.RequestEntityAcceptance}.
      */
-    @Deprecated
-    public abstract akka.http.scaladsl.model.RequestEntityAcceptance requestEntityAcceptance();
+    public abstract akka.http.javadsl.model.RequestEntityAcceptance requestEntityAcceptance();
```
Comment (raboof) [inline akka-http-core/src/main/java/akka/http/javadsl/model/HttpMethod.java]: Subtle - so if you'd want to support both akka-http 10.0 and 10.1 you'd have to use the now-deprecated `getRequestEntityAcceptance` - that makes sense and is in line with the comment on the 'old' `requestEntityAcceptance()`, so 👍

File: akka-http-core/src/main/java/akka/http/javadsl/model/HttpMethods.java [hunk 2]
```diff
@@ -27,18 +27,17 @@ private HttpMethods() {}
 
     /**
      * Create a custom method type.
-     * @deprecated Use {@link  #createCustom} instead.
+     * @deprecated Use {@link #custom} instead. Will be removed in Akka HTTP 11.0.
      */
     @Deprecated
-    public static HttpMethod custom(String value, boolean safe, boolean idempotent, akka.http.scaladsl.model.RequestEntityAcceptance requestEntityAcceptance) {
-        return akka.http.scaladsl.model.HttpMethod.custom(value, safe, idempotent, requestEntityAcceptance);
+    public static HttpMethod createCstom(String value, boolean safe, boolean idempotent, akka.http.javadsl.model.RequestEntityAcceptance requestEntityAcceptance) {
```
Comment (raboof) [inline akka-http-core/src/main/java/akka/http/javadsl/model/HttpMethods.java]: Typo: missing 'u'

File: akka-http-core/src/main/java/akka/http/javadsl/model/headers/HttpChallenge.java [hunk 3]
```diff
@@ -17,22 +17,6 @@
 
     public abstract Map<String, String> getParams();
 
-    /**
-     * @deprecated Use constructor with optional realm parameter instead.
-     */
-    @Deprecated
-    public static HttpChallenge create(String scheme, String realm) {
```
Comment (jrudolph) [inline akka-http-core/src/main/java/akka/http/javadsl/model/headers/HttpChallenge.java]: Do we know what's the reasoning behind deprecating those? They seem like useful auxiliary constructors.

File: akka-http-core/src/test/scala/akka/http/javadsl/JavaInitializationSpec.scala [hunk 4]
```diff
@@ -1,66 +0,0 @@
...
-    def =!=(expected: String) = {
-      self should !==(null)
-      self.toString shouldBe expected
-    }
-  }
-
-  "EntityTagRange" should {
-    "initializes the right field" in {
-      akka.http.scaladsl.model.headers.EntityTagRange.`*` =!= "*"
-      akka.http.javadsl.model.headers.EntityTagRanges.ALL =!= "*"
```
Comment (jrudolph) [inline akka-http-core/src/test/scala/akka/http/javadsl/JavaInitializationSpec.scala]: Shouldn't we keep testing the constants that still work?

File: akka-http-core/src/main/java/akka/http/javadsl/model/headers/HttpChallenge.java (hunk 3 above)
Comment (ktoso) [reply akka-http-core/src/main/java/akka/http/javadsl/model/headers/HttpChallenge.java]: The idea was to "move to Paths everywhere", but I don't mind keeping them

File: akka-http-core/src/test/scala/akka/http/javadsl/JavaInitializationSpec.scala (hunk 4 above)
Comment (ktoso) [reply akka-http-core/src/test/scala/akka/http/javadsl/JavaInitializationSpec.scala]: This was some crazy edge case so tests best to keep AFAICS

**Comment by jonas** [comment]
I've addressed all comments.

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala [hunk 5]
```diff
@@ -278,8 +276,7 @@ object HttpEntity {
    *
    * If the given `chunkSize` is -1 the default chunk size is used.
    */
-  @deprecated("Use `fromPath` instead", "2.4.5")
-  def apply(contentType: ContentType, file: File, chunkSize: Int = -1): UniversalEntity =
+  def fromFile(contentType: ContentType, file: File, chunkSize: Int = -1): UniversalEntity =
```
Comment (jrudolph) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala]: :+1:

File: akka-http-core/src/test/scala/akka/http/javadsl/JavaInitializationSpec.scala [hunk 6]
```diff
@@ -19,46 +19,40 @@ class JavaInitializationSpec extends WordSpec with Matchers {
     "initializes the right field" in {
       akka.http.scaladsl.model.headers.EntityTagRange.`*` =!= "*"
       akka.http.javadsl.model.headers.EntityTagRanges.ALL =!= "*"
-      akka.http.javadsl.model.headers.EntityTagRange.ALL =!= "*"
```
Comment (jrudolph) [inline akka-http-core/src/test/scala/akka/http/javadsl/JavaInitializationSpec.scala]: :+1:

Weird that they weren't all in the same order because the bug so much depended on the order of initialization...

File: akka-http-core/src/main/java/akka/http/javadsl/model/headers/HttpChallenge.java [hunk 7]
```diff
@@ -17,20 +17,12 @@
 
     public abstract Map<String, String> getParams();
 
-    /**
-     * @deprecated Use constructor with optional realm parameter instead.
-     */
-    @Deprecated
     public static HttpChallenge create(String scheme, String realm) {
```
Comment (jrudolph) [inline akka-http-core/src/main/java/akka/http/javadsl/model/headers/HttpChallenge.java]: :+1:
